lib/facter/warranty.rb

```diff
@@ -17,40 +17,72 @@ def create_dell_warranty_cache(cache)
   start_date      = Time.now() # Push start time back and expiration forward
   servicetag      = Facter.value('serialnumber')
 
+  # Dell v5 API requires OAuth2 setup before the call, but this just adds one minor step
+  # (ends up with the same data in the response, in a very similar format to v4 responses!)
+
   begin
-    # rescue in case dell.com is down
-    dell_api_key     = File.read("/etc/dell_api_key") # Production API key needs to be grabbed from a file
-    uri              = URI.parse("https://api.dell.com/support/assetinfo/v4/getassetwarranty/#{servicetag}?apikey=#{dell_api_key}")
+    dell_client_id     = File.read("/etc/dell_client_id")     # API v5 client_id needs to be grabbed from a file
+    dell_client_secret = File.read("/etc/dell_client_secret") # API v5 client_secret needs to be grabbed from a file
```

Comment:
I'd rather have a single file for that setting, so maybe `/etc/dell_warranty.conf` with `id` and `secret` entries in there